data/tutorials/language/0lg_00_basic_datatypes.md

```diff
@@ -0,0 +1,808 @@
+```ocaml
+# 'a';;
+- : char = 'a'
+```
+Operations on `char` values are provided by the [`Stdlib`](/api/Stdlib.html) and the [`Char`](/api/Char.html) modules.
+
+The module [`Uchar`](/api/Uchar.html) provides support for Unicode characters.
+
+### Strings & Byte Sequences
+
+#### Strings
+
+Strings are finite and fixed-sized sequences of `char` values. Strings are immutable, meaning it is impossible to change a character's value inside a string. The string concatenation operator symbol is `^`.
+```ocaml
+# "hello" ^ " " ^ "world!";;
+- : string = "hello world!"
+```
+
+Indexed access to string characters is possible using the following syntax:
+```ocaml
+# "buenos dias".[4];;
+- : char : 'o'
+```
+Operations on `string` values are provided by the [`Stdlib`](/api/Stdlib.html) and the [`String`](/api/String.html) modules.
+
+#### Byte Sequences
+
+Byte sequences are finite and fixed-sized. Each individual byte is represented by a `char` value. Byte sequences are mutable, meaning they can't be extended or shortened, but each component byte may be updated. Essentially, a byte sequence (type `bytes`) is a mutable string that can't be printed. There is no way to write `bytes` literally, so they must be produced by a function.
+```ocaml
+# String.to_bytes "hello";;
+- : bytes = Bytes.of_string "hello"
+```
+Operations on `bytes` values are provided by the [`Stdlib`](/api/Stdlib.html) and the [`Bytes`](/api/Bytes.html) modules. Only the function `Bytes.get` allows direct access to the characters contained in a byte sequence. There is no direct access operator on byte sequences.
+
+### Arrays & Lists
+
+#### Arrays
+
+Arrays are finite and fixed-sized sequences of values of the same type. Here are a couple of examples:
+```ocaml
+# [| 0; 1; 2; 3; 4; 5 |];;
+- : int array = [|0; 1; 2; 3; 4; 5|]
+
+# [| 'x'; 'y'; 'z' |];;
+- : char array = [|'x'; 'y'; 'z'|]
+
+# [| "foo"; "bar"; "baz" |];;
+- : string array = [|"foo"; "bar"; "baz"|]
+```
+
+Arrays may contain values of any type. Here arrays are `int array`, `char array`, and `string array`, but any type of data can be used in an array. Usually, `array` is said to be a polymorphic type. Strictly speaking, it is a type operator, and it accepts a type as a parameter (here `int`, `char`, and `string`) to form another type (those inferred here). This is the empty array.
+```ocaml
+# [||];;
+- : 'a array = [||]
+```
+Remember, `'a` ("alpha") is a type variable: a type parameter that will be replaced by another type.
+
+Like `string` and `bytes`, arrays support direct access, but the syntax is not the same.
+```ocaml
+# [| 'x'; 'y'; 'z' |].(2);;
+- : char = 'z'
+```
+
+Arrays are mutable, meaning they can't be extended or shortened, but each element may be updated.
+```ocaml
+# let letter = [| 'v'; 'x'; 'y'; 'z' |];;
+val letter : char array = [|'v'; 'x'; 'y'; 'z'|]
+
+# letter.(2) <- '3';;
```

Comment:
Since the arrow is pointing the other way for the first time (that I remember), should we speak to it? After some pondering, I saw what it was doing (passing `3` to the 2nd position of `letter`, but it confused me at first.

Reply:
I'm writing something on this.